From 97515341454f26fcd14e50d5980f5f0361307814 Mon Sep 17 00:00:00 2001
From: FlorianPalm <f.palm@plt.rwth-aachen.de>
Date: Mon, 12 Jan 2015 18:45:01 +0100
Subject: [PATCH] fixes to the patch

---
 CMakeLists.txt | 19 ++++++++-----------
 src/ua_util.h  |  2 +-
 2 files changed, 9 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c785182..9e66da0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,19 +47,16 @@ set(lib_sources src/ua_types.c
 
 # compiler flags
 if(CMAKE_COMPILER_IS_GNUCC OR "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
-add_definitions(-std=c99 -pedantic -pipe -Wall -Wextra -Werror -Wformat
-                -Wno-unused-parameter -Wno-unused-function -Wno-unused-label -Wpointer-arith -Wreturn-type -Wsign-compare -Wmultichar
-                -Wshadow -Wcast-align -Wcast-qual -Wmissing-prototypes -Wstrict-prototypes # -Wconversion
-                -Winit-self -Wuninitialized -Wno-deprecated -Wformat-security -ffunction-sections -fdata-sections)
+add_definitions(-std=c99 -fomit-frame-pointer -pipe -msoft-float -fno-builtin -DEMBED -fno-strict-aliasing)
     if(NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
         add_definitions(-Wformat-nonliteral)
-        set (CMAKE_C_LINK_FLAGS "${CMAKE_C_LINK_FLAGS} -Wl,--gc-sections")
-        set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--gc-sections")
+        set (CMAKE_C_LINK_FLAGS "${CMAKE_C_LINK_FLAGS} -Wl,-elf2flt -msoft-float")
+        set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}  -Wl,-elf2flt -msoft-float")
     else()
         add_definitions(-Wno-gnu-statement-expression)
     endif()
 	if(NOT WIN32)
-	    add_definitions(-fstack-protector -fPIC -fvisibility=hidden)
+	   # add_definitions(-fPIC)
 	endif()
 endif()
 
@@ -150,8 +147,8 @@ else()
 endif()
 
 add_library(open62541-objects OBJECT ${lib_sources}) # static version that exports all symbols
-add_library(open62541 SHARED $<TARGET_OBJECTS:open62541-objects>)
-target_compile_definitions(open62541-objects PUBLIC UA_DYNAMIC_LINKING)
+add_library(open62541 $<TARGET_OBJECTS:open62541-objects>)
+#target_compile_definitions(open62541-objects PUBLIC UA_DYNAMIC_LINKING)
 
 ## logging
 set(UA_LOGLEVEL 400 CACHE STRING "Level at which logs shall be reported")
@@ -170,7 +167,7 @@ endif()
 configure_file("src/ua_config.h.in" "${PROJECT_BINARY_DIR}/src_generated/ua_config.h")
 
 # download queue.h if required
-if(WIN32)
+#if(WIN32)
     if(NOT EXISTS "${PROJECT_BINARY_DIR}/src_generated/queue.h")
         file(DOWNLOAD "http://openbsd.cs.toronto.edu/cgi-bin/cvsweb/~checkout~/src/sys/sys/queue.h" "${PROJECT_BINARY_DIR}/src_generated/queue.h" STATUS result)
         list(GET result 0 download_ok)
@@ -179,7 +176,7 @@ if(WIN32)
             message(FATAL_ERROR "queue.h could not be downloaded")
         endif()
     endif()
-endif(WIN32)
+#endif(WIN32)
 
 # generate code from xml definitions
 file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/src_generated")
diff --git a/src/ua_util.h b/src/ua_util.h
index 5ec1191..65e1a4d 100644
--- a/src/ua_util.h
+++ b/src/ua_util.h
@@ -5,7 +5,7 @@
 
 #include <stddef.h> /* Needed for sys/queue.h */
 #ifndef WIN32
-#include <sys/queue.h>
+#include "queue.h"
 #include <alloca.h>
 #else
 #include "queue.h"
-- 
2.2.0

